### PR TITLE
Fix regression from GCM mode refactoring

### LINKED ISSUE #18896: Regression in AES-GCM on Alpha and Arm platforms after the GCM refactoring
The on-push cross-compile builds in CI for Alpha and Arm platforms fail since the GCM refactoring was merged.

See: https://github.com/openssl/openssl/runs/7468446782?check_suite_focus=true https://github.com/openssl/openssl/runs/7468446918?check_suite_focus=true

Perhaps something like https://github.com/openssl/openssl/commit/48e35b99bd0071207cfe39da22eb2502db5c09dc will be needed?

### PR DESCRIPTION
**Pull request by t8m**
Fixes #18896

### CODE REVIEW

**Review by tmshort — APPROVED** [review]
Optimally, the initialization of these at the top of `gcm_get_funcs()` would've prevented this, but it's just too complex.
Because this function is used by the AES-GCM-SIV PR, there should be no changes needed there.

**Review by paulidale — APPROVED** [review]
LGTM, the test will be once merged :)

**Comment by paulidale** [comment]
Since this is fixing a CI, I think it qualifies as urgent and have merged it to master.

Merging it will also tell if it succeeds at fixing the issue which is seems to have do.

**Pull request closed without merging**
